docs/paper/dev/api/commands-api/arguments.mdx

```diff
@@ -0,0 +1,190 @@
+                            );
+                            return Command.SINGLE_SUCCESS;
+                        })
+                ).build()
+            );
+        });
+    }
+}
+```
+This command has one argument of the `Enchantment` datatype. The advantage of using
+Brigadier over the `TabExecutor` interface is that you don't have to convert the
+string that is passed to the `CommandExecutor#onCommand()` function as an argument to 
+an enchantment in this case because `ctx.getArgument("enchantmentargument", Enchantment.class)`
+returns a value of the `Enchantment` datatype. Additionally, you don't have to write 
+error handling functionality yourself as the client checks whether the argument the
+user wrote in the chat is one of the `minecraft:enchantment` type and if it isn't,
+the user will get an error. Another advantage `ArgumentTypes` have over string-based argument systems like
+Bukkit's `TabExecutor` is that we can work with our own types and the completions list 
+is not always sorted alphanumerically. This may sound insignificant but when you want an input 
+of the type integer and suggest the numbers 0-10, for example, your list will look like this because
+it needs to be converted to strings:`0, 1, 10, 2, 3, ...`. Looks awful, right?
+
+## Enchantment types
+
+By default, you can use [registries](../registries) to get simple argument types like
+blocks, items, potions and many more. In the example above, we used the Enchantment 
+Argument type but there are many others:
+
+### Predefined types (Registry)
+
+| Registry key name   | Return datatype class | Description                                                                                                                                                                                                                                                                                  |
+|---------------------|-----------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| GAME_EVENT          | GameEvent             | Events in the game (eating, flying with an elytra etc.)                                                                                                                                                                                                                                      |
+| STRUCTURE_TYPE      | StructureType         | [Structures](https://minecraft.wiki/w/Structure#Overworld) - Beware that, as of Paper 1.20.6-147, there is also a Registry named `STRUCTURE` that is not recognized by the client and makes it unable to connect. This has been reported to Paper developers and will be fixed in a future release. |
```

Comment:
presuming this is referring to https://github.com/PaperMC/Paper/issues/10991; this bug hasn't been triaged yet and I don't think it needs to be documented here, as it will likely be promptly fixed and the comment obsoleted